src/clj/game/core/gaining.clj

```diff
@@ -45,47 +46,49 @@
                    (pos? (get-in @state [:runner :run-credit] 0)))
           (swap! state update-in [:runner :run-credit] (sub->0 value))))))
 
-(defn gain [state side & args]
-  (doseq [[cost-type amount] (partition 2 args)]
-    (cond
-      ;; amount is a map, merge-update map
-      (map? amount)
-      (doseq [[subtype amount] amount]
-        (swap! state update-in [side cost-type subtype] (safe-inc-n amount))
-        (swap! state update-in [:stats side :gain cost-type subtype] (fnil + 0) amount))
+(defn gain [state side eid & args]
+  (effect-seq state side eid [[cost-type amount] (partition 2 args)]
+    (let [continuation #(trigger-event-simult state side eid (if (= side :corp) :corp-gain :runner-gain) nil [cost-type amount])]
```

Comment:
I'm not a fan of this kind of code reuse. Makes it harder to see what's happening in each sub-form.